cmake/modules/FindD3D12.cmake

```diff
@@ -1,7 +1,18 @@
 # Find the Win10 SDK path.
 if ("$ENV{WIN10_SDK_PATH}$ENV{WIN10_SDK_VERSION}" STREQUAL "" )
   get_filename_component(WIN10_SDK_PATH "[HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Windows Kits\\Installed Roots;KitsRoot10]" ABSOLUTE CACHE)
-  set (WIN10_SDK_VERSION ${CMAKE_VS_WINDOWS_TARGET_PLATFORM_VERSION})
+  if (CMAKE_VS_WINDOWS_TARGET_PLATFORM_VERSION)
+    set (WIN10_SDK_VERSION ${CMAKE_VS_WINDOWS_TARGET_PLATFORM_VERSION})
+  else()
+    # CMAKE_VS_WINDOWS_TARGET_PLATFORM_VERSION may not be defined if, for example,
+    # the Ninja generator is used instead of Visual Studio. Attempt to retrieve the
+    # most recent SDK version from the list of paths under "${WIN10_SDK_PATH}/Include/".
+    file(GLOB sdk_dirs RELATIVE "${WIN10_SDK_PATH}/Include/" "${WIN10_SDK_PATH}/Include/10.*")
+    if (sdk_dirs)
+      list(POP_BACK sdk_dirs WIN10_SDK_VERSION)
+    endif()
+    unset(sdk_dirs)
```

Comment:
Note that this makes use of `file(GLOB...)` to get the list of directories from the file system. There is a way to actually get this list from the registry, but it requires CMake 3.24 to make use of the [windows registry query function to get the list of subkeys](https://cmake.org/cmake/help/latest/command/cmake_host_system_information.html#query-windows-registry). The code would look like this:

```
  # Using the Ninja generator will not set CMAKE_VS_WINDOWS_TARGET_PLATFORM_VERSION.
  # Attempt to retrieve the most recent SDK version by retrieving the last subkeys under "Installed Roots"
  cmake_host_system_information(RESULT INSTALLED_ROOTS QUERY WINDOWS_REGISTRY "HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Windows Kits\\Installed Roots" SUBKEYS ERROR_VARIABLE ERROR_RESULT)
  list(POP_BACK INSTALLED_ROOTS WIN10_SDK_VERSION)
```

But since DXC requires CMake 3.17.2, I went with file globbing instead.

Reply:
In CMake 3.17 (and many earlier versions) there is an undocumented feature of `get_filename_component` to read registry values. We actually use that in this file a few lines above your change.

Reply:
> In CMake 3.17 (and many earlier versions) there is an undocumented feature of `get_filename_component` to read registry values. We actually use that in this file a few lines above your change.

Yeah, but unfortunately it cannot be used to read subkeys. I looked at the source code for CMake to see if it was possible.